### PR TITLE
Collapse _like overloads into a single overload.

### PR DESCRIPTION
**Pull request by ezyang**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #33826 Stop filling in defaults in Python binding layer.
* **#33518 Collapse _like overloads into a single overload.**
* #33516 Add MemoryFormat to TensorOptions, but not codegen.
* #33825 Turn on strict dtype checking for test_torch.py

The fact that there were two overloads appears to be a historical
artifact that dates back to when goldsborough originally added these
bindings in the first place.  If TensorOptions is made optional,
then you only need one overload, not two, as they are exactly redundant
with each other.  When MemoryFormat was added, it was made a little
harder to do this, as the C++ syntax at::empty_like(t, memory_format) would
not work if you collapsed the overload; but now it works because TensorOptions
supports MemoryFormat.

The upshot is, I can get rid of all the overloads and just have one overload.
Amazingly, this change is backwards compatible, as the test attests.  While
I was at it, I also deleted the overload name from the functions entirely.

Signed-off-by: Edward Z. Yang <ezyang@fb.com>

Differential Revision: [D20034385](https://our.internmc.facebook.com/intern/diff/D20034385)

### CODE REVIEW

File: aten/src/ATen/native/TensorFactories.cpp [hunk 1]
```diff
@@ -196,9 +196,8 @@ AT_FORALL_SCALAR_TYPES_AND3(Bool, Half, BFloat16, DEFINE_CAST_OP)
 #undef DEFINE_CAST_OP
 
 Tensor empty_like(
```
Comment (VitalyFedyunin) [inline aten/src/ATen/native/TensorFactories.cpp]: If we are collapsing, do we still need `native::empty_like(tensor)`?

Comment (ezyang) [reply aten/src/ATen/native/TensorFactories.cpp]: No, these can be deleted.

**Review by suo — APPROVED** [review]
lgtm, check on the relevant fbcode backcompat tests just to be sure!

**Review by VitalyFedyunin — APPROVED** [review]
Thanks for adding more jit tests

**Pull request closed without merging**